platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StaticPlatformBuilder.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform;
+
+import static com.swirlds.logging.LogMarker.STARTUP;
+import static com.swirlds.platform.util.BootstrapUtils.startJVMPauseDetectorThread;
+import static com.swirlds.platform.util.BootstrapUtils.startThreadDumpGenerator;
+import static com.swirlds.platform.util.BootstrapUtils.writeSettingsUsed;
+
+import com.swirlds.common.config.PathsConfig;
+import com.swirlds.common.config.singleton.ConfigurationHolder;
+import com.swirlds.common.metrics.Metrics;
+import com.swirlds.common.metrics.platform.DefaultMetricsProvider;
+import com.swirlds.common.startup.Log4jSetup;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.logging.payloads.NodeStartPayload;
+import com.swirlds.platform.util.BootstrapUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Static setup code for a platform. Logic in this class should gradually disappear as we move away from the use of
+ * static stateful objects.
+ */
+final class StaticPlatformBuilder {
```

Comment:
From my point of view the class should directly be deprecated

Reply:
There is a very small amount of logic that I do not think we will ever remove.

- starting the thread dump generator
- starting the JVM pause detector
- writing the "hello world" log message (this may change with logging contexts)
- system health checks